scripts/prepdocs/files.py

```diff
@@ -0,0 +1,749 @@
+                page_length = page.spans[0].length
+                table_chars = [-1] * page_length
+                for table_id, table in enumerate(tables_on_page):
+                    for span in table.spans:
+                        # replace all table spans with "table_id" in table_chars array
+                        for i in range(span.length):
+                            idx = span.offset - page_offset + i
+                            if idx >= 0 and idx < page_length:
+                                table_chars[idx] = table_id
+
+                # build page text by replacing characters in table spans with table html
+                page_text = ""
+                added_tables = set()
+                for idx, table_id in enumerate(table_chars):
+                    if table_id == -1:
+                        page_text += form_recognizer_results.content[page_offset + idx]
+                    elif table_id not in added_tables:
+                        page_text += DocumentAnalysisPdfParser.table_to_html(tables_on_page[table_id])
+                        added_tables.add(table_id)
+
+                yield Page(page_num=page_num, offset=offset, text=page_text)
+                offset += len(page_text)
+
+    @classmethod
+    def table_to_html(cls, table: DocumentTable):
+        table_html = "<table>"
+        rows = [
+            sorted([cell for cell in table.cells if cell.row_index == i], key=lambda cell: cell.column_index)
+            for i in range(table.row_count)
+        ]
+        for row_cells in rows:
+            table_html += "<tr>"
+            for cell in row_cells:
+                tag = "th" if (cell.kind == "columnHeader" or cell.kind == "rowHeader") else "td"
+                cell_spans = ""
+                if cell.column_span > 1:
+                    cell_spans += f" colSpan={cell.column_span}"
+                if cell.row_span > 1:
+                    cell_spans += f" rowSpan={cell.row_span}"
+                table_html += f"<{tag}{cell_spans}>{html.escape(cell.content)}</{tag}>"
+            table_html += "</tr>"
+        table_html += "</table>"
+        return table_html
+
+
+class BlobManager:
+    def __init__(
+        self,
+        endpoint: str,
+        container: str,
+        credential: Union[AsyncTokenCredential, AzureKeyCredential],
+        verbose: bool = False,
+    ):
+        self.endpoint = endpoint
+        self.credential = credential
+        self.container = container
+        self.verbose = verbose
+
+    async def upload_blob(self, file: File):
+        async with BlobServiceClient(
+            account_url=self.endpoint, credential=self.credential
+        ) as service_client, service_client.get_container_client(self.container) as container_client:
+            if not await container_client.exists():
+                await container_client.create_container()
+
+                # if file is PDF split into pages and upload each page as a separate blob
```

Comment:
This is the logic I changed in my PR which I can either merge now or redo based on this change. It actually reduces the code complexity.